src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -303,6 +307,20 @@ public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapInd
             return clone;
         }
 
+        private ArrowStringColumn Clone(PrimitiveColumn<bool> boolColumn)
+        {
+            if (boolColumn.Length > Length)
+                throw new ArgumentException(Strings.MapIndicesExceedsColumnLenth, nameof(boolColumn));
+            ArrowStringColumn ret = new ArrowStringColumn(Name);
+            for (long i = 0; i < boolColumn.Length; i++)
+            {
+                bool? value = boolColumn[i];
+                if (value.HasValue && value.Value == true)
+                    ret.Append(this[i]);
```

Comment:
This is less than ideal, since you are going from UTF8 => UTF16 => UTF8. Probably want to optimize this to not have to encode/decode the characters.